### DISCUSSION TITLE
What is the best aproach of including `meta` tags only in blog-post pages?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#7701] [Q&A] [answered]
I want to include specific Meta html tags in blog post's `<head>` but I'm not sure what the best aproach would be.

I first thought I could extend the `blog-post.html` file and override the `site_meta` block like this:
```jinja2
{% extends "main.html" %}

{% block site_meta %}
  {{ super() }} <!-- Make sure we include previous meta stuff -->
  <meta name="lorem" content="ipsum">
{% endblock site_meta %}
```
and while this seems to include the tag as wanted, does the page change. Namely, the infobox on the left-hand side is completely gone. Trying to fix this, i.e. by also importing nav items and `super()` the container block, didn't work so far and I'm unsure what other aproaches I could try here.

To be clear: My goal is to have a meta tag only in blog posts, but no other pages.

**Comment by kamilkrzyskow** [comment]
Hello @Andre601,
the code above should work, when you put it in the `blog-post` template then only the blog posts (files that use this template) will use it.
Only thing weird about it compared to the docs here, is the `endblock site_meta` with the name of the block. The super() should preserve the previous content.

If it's not working please provide a full example, so that we can reproduce it.

Reference on how to do:
- https://squidfunk.github.io/mkdocs-material/reference/#on-a-single-page

**Reply by kamilkrzyskow** [reply, marked as answer]
If that's the only thing you're changing in the blog-post.html, you could just forgo the file override and just override the site_meta block inside your `main.html`. You should be able too access variables like `{{ page.__class__.__module__ }}` or `{{ page.__class__.__name__ }}` to detect what is the Page's class, and then you'd use it in a conditional to limit the changes accordingly.

Or without Python shenanigans, just access the `page.meta.template` to find out what the template name is and check if it's `blog-post.html` 😅

**Reply by Andre601** [reply]
Well, my example is everything I use for `blog-post.html`

I can try the aproach with `main.html`. Already extend it for more meta content, so it wouldn't be such a big change.

**Reply by Andre601** [reply]
Okay, the aproach with `main.html` works. Tho, not sure why exactly the aproach in `blog-post.html` didn't work for me...
Maybe I did it in the wrong section or smth? Because looking at `main.html`, I can now see that I override the `extrahead` block and not `site_meta`

**Reply by Andre601** [reply]
I think the main issue is that `{{ super() }}` isn't working for the `container` block. Probably due to lack of some important stuff (maybe related to nav-items?) or context... Copying the entire content of the block as-is over fixes the missing block on the left side of the page.

**Reply by kamilkrzyskow** [reply]
Ah, I glossed over this part being the issue:
> and while this seems to include the tag as wanted, does the page change. Namely, the infobox on the left-hand side is completely gone.

and focused only on:
> My goal is to have a meta tag only in blog posts, but no other pages.

getting this to work, sorry about that.

---

You can run `mkdocs serve --watch-theme` and it will show all of the "theme" paths, so in order:
- your project `overrides`, 
- the material theme, 
- and mkdocs default theme. 

**All of this is "merged" in order to represent one single file tree.**

Each `page.meta.template` is the entry point file for the templating of the rendered page. The blog plugins sets the template of the blog posts to `blog-post.html`, and the default logic is here:
- https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/blog-post.html
- https://github.com/squidfunk/mkdocs-material/blob/master/material/templates/blog-post.html

However, if you override the `blog-post.html` it will replace the file provided by the material theme. If you only make a `blog-post.html` containing the `site_meta` block you remove the `container` block override that supplies the custom left-hand side panel.

> Copying the entire content of the block as-is over fixes the missing block on the left side of the page.

So that's the expected and correct action, you have to fully copy the file from the source and only make adjustments. 

The `blog-post.html` removes the other navigation as it would collide with the custom left panel, so it fully replaces the `container` block without the use of `super()`. The `super()` refers to the parent and at the top of `blog-post.html` there is `{% extends "main.html" %}` and your `main.html` extends `base.html`, each part of the chain inherits blocks from the parent, and you can include their content with `super()` or omit it to fully replace the contents of a given block.

It just occurred to me that `blog-post.html` is both the entry point, but also the last to "render" as it extends all other stuff from both `main.html` and `base.html`. So the template is both the start and the end 🤯

**Reply by Andre601** [reply]
Okay, that explains the issue. Thanks.
